admiral/pkg/clusters/serviceentry.go

```diff
@@ -147,8 +154,31 @@ func modifyServiceEntryForNewServiceOrPod(event admiral.EventType, env string, s
 			for _, ep := range serviceEntry.Endpoints {
 				//replace istio ingress-gateway address with local fqdn, note that ingress-gateway can be empty (not provisoned, or is not up)
 				if ep.Address == clusterIngress || ep.Address == "" {
-					// see if we have weighted services (rollouts with canary strategy)
-					if len(weightedServices) > 1 {
+					// Update endpoints with locafqdn for active and preview se of bluegreen rollout
+					if isBlueGreenStrategy {
+						activeServiceName := sourceRollouts[sourceCluster].Spec.Strategy.BlueGreen.ActiveService
```

Comment:
Abstract all this logic into a method similar to `updateEndpointsForWeightedServices`